### PR TITLE
Improve formatting of decimal labels in WCSAxes

### PR DESCRIPTION
**Pull request by astrofrog**
This makes the following improvements to WCSAxes:

* ``CoordinateHelper.set_format_unit`` now has ``decimal`` and ``show_decimal_unit`` options to set whether to use decimal or sexagesimal formatting by default for the axis, and whether to show units in the decimal case.

* Units are now shown by default for decimal formatting (this is a change from before - while it will change the output, this won't **break** code as other API changes might)

* Improves the determination of the tick spacing in the decimal case

Here is the output of:

```python
fig = plt.figure(figsize=(12, 15))
for i, unit in enumerate(['degree', 'hour', 'hourangle', 'arcmin', 'arcsec', 'mas']):
    ax = fig.add_subplot(3, 2, i + 1, projection=wcs)
    ax.coords[0].set_format_unit(unit, decimal=True)
    ax.set_title(f'unit={unit}')
plt.savefig('decimal.png', dpi=200)
```

decimal

and with ``rcParams['text.usetex'] = True``:

decimal_latex

This still needs:

* [x] Tests
* [x] A changelog entry

But before I do this, I wanted to check if people are on board with this?

I think 52e7c86 should probably be backported to 2.0.x and 3.0.x, so I might do manual PRs with just that commit to those branches.

cc @Cadair @larrybradley @lpsinger

### CODE REVIEW

**Comment by lpsinger** [comment]
LGTM

**Comment by larrybradley** [comment]
LGTM too.  One minor comment -- should there be a space before ("mas")?

**Comment by astrofrog** [comment]
> LGTM too. One minor comment -- should there be a space before ("mas")?

This is the way astropy.units formats quantities by default but maybe it does make sense to insert a space.

**Comment by Cadair** [comment]
Seems to work with the default sunpy plots, I will do a little more testing.

**Comment by Cadair** [comment]
This seems to still work for the two main solar celestial systems.

**Comment by astrofrog** [comment]
@larrybradley - for the space before mas, this is going to require making it possible to do that in Angle.to_string, so I've opened an issue here: https://github.com/astropy/astropy/issues/7455

**Comment by astrofrog** [comment]
@larrybradley @lpsinger @Cadair - this is ready for a final review!

File: astropy/visualization/wcsaxes/utils.py [hunk 1]
```diff
@@ -129,8 +129,10 @@ def coord_type_from_ctype(ctype):
     elif ctype[1:4] == 'LON':
         return 'longitude', None, None
     elif ctype[:4] in ['HPLN']:
-        return 'longitude', None, 180.
-    elif ctype[:4] in ['DEC-', 'HPLT'] or ctype[1:4] == 'LAT':
+        return 'longitude', u.arcsec, 180.
```
Comment (Cadair) [inline astropy/visualization/wcsaxes/utils.py]: can you also add a:
```python
if cytpe[:4] in ['HGLN']:
    return 'longitude', None, 180.
```

in here as well (it's conventionally in decimal degrees so I assume `None` is ok?)

**Comment by larrybradley** [comment]
This PR needs to be rebased now. :smile:

**Pull request merged**